cmd/snap-seccomp/main_test.go

```diff
@@ -0,0 +1,404 @@
+		}
+	}
+}
+
+// ported from test_restrictions_working_args_quotactl
+func (s *snapSeccompSuite) TestRestrictionsWorkingArgsQuotactl(c *C) {
+	bpf.VmEndianness = nativeEndian()
+
+	for _, arg := range []string{"Q_QUOTAON", "Q_QUOTAOFF", "Q_GETQUOTA", "Q_SETQUOTA", "Q_GETINFO", "Q_SETINFO", "Q_GETFMT"} {
+		seccompWhitelist := fmt.Sprintf("quotactl %s", arg)
+		bpfInputGood := fmt.Sprintf("quotactl;native;%s", arg)
+		simulateBpf(c, seccompWhitelist, bpfInputGood, main.SeccompRetAllow)
+		for _, bad := range []string{"quotactl;native;99999", "read;native;"} {
+			simulateBpf(c, seccompWhitelist, bad, main.SeccompRetKill)
+		}
+	}
+}
+
+// ported from test_restrictions_working_args_prctl
+func (s *snapSeccompSuite) TestRestrictionsWorkingArgsPrctl(c *C) {
+	bpf.VmEndianness = nativeEndian()
+
+	for _, arg := range []string{"PR_CAP_AMBIENT", "PR_CAP_AMBIENT_RAISE", "PR_CAP_AMBIENT_LOWER", "PR_CAP_AMBIENT_IS_SET", "PR_CAP_AMBIENT_CLEAR_ALL", "PR_CAPBSET_READ", "PR_CAPBSET_DROP", "PR_SET_CHILD_SUBREAPER", "PR_GET_CHILD_SUBREAPER", "PR_SET_DUMPABLE", "PR_GET_DUMPABLE", "PR_SET_ENDIAN", "PR_GET_ENDIAN", "PR_SET_FPEMU", "PR_GET_FPEMU", "PR_SET_FPEXC", "PR_GET_FPEXC", "PR_SET_KEEPCAPS", "PR_GET_KEEPCAPS", "PR_MCE_KILL", "PR_MCE_KILL_GET", "PR_SET_MM", "PR_SET_MM_START_CODE", "PR_SET_MM_END_CODE", "PR_SET_MM_START_DATA", "PR_SET_MM_END_DATA", "PR_SET_MM_START_STACK", "PR_SET_MM_START_BRK", "PR_SET_MM_BRK", "PR_SET_MM_ARG_START", "PR_SET_MM_ARG_END", "PR_SET_MM_ENV_START", "PR_SET_MM_ENV_END", "PR_SET_MM_AUXV", "PR_SET_MM_EXE_FILE", "PR_MPX_ENABLE_MANAGEMENT", "PR_MPX_DISABLE_MANAGEMENT", "PR_SET_NAME", "PR_GET_NAME", "PR_SET_NO_NEW_PRIVS", "PR_GET_NO_NEW_PRIVS", "PR_SET_PDEATHSIG", "PR_GET_PDEATHSIG", "PR_SET_PTRACER", "PR_SET_SECCOMP", "PR_GET_SECCOMP", "PR_SET_SECUREBITS", "PR_GET_SECUREBITS", "PR_SET_THP_DISABLE", "PR_TASK_PERF_EVENTS_DISABLE", "PR_TASK_PERF_EVENTS_ENABLE", "PR_GET_THP_DISABLE", "PR_GET_TID_ADDRESS", "PR_SET_TIMERSLACK", "PR_GET_TIMERSLACK", "PR_SET_TIMING", "PR_GET_TIMING", "PR_SET_TSC", "PR_GET_TSC", "PR_SET_UNALIGN", "PR_GET_UNALIGN"} {
+		seccompWhitelist := fmt.Sprintf("prctl %s", arg)
+		bpfInputGood := fmt.Sprintf("prctl;native;%s", arg)
+		simulateBpf(c, seccompWhitelist, bpfInputGood, main.SeccompRetAllow)
+		for _, bad := range []string{"prctl;native;99999", "read;native;"} {
+			simulateBpf(c, seccompWhitelist, bad, main.SeccompRetKill)
+		}
+
+		if arg == "PR_CAP_AMBIENT" {
+			for _, j := range []string{"PR_CAP_AMBIENT_RAISE", "PR_CAP_AMBIENT_LOWER", "PR_CAP_AMBIENT_IS_SET", "PR_CAP_AMBIENT_CLEAR_ALL"} {
+				seccompWhitelist := fmt.Sprintf("prctl %s %s", arg, j)
+				bpfInputGood := fmt.Sprintf("prctl;native;%s,%s", arg, j)
+				simulateBpf(c, seccompWhitelist, bpfInputGood, main.SeccompRetAllow)
+				for _, bad := range []string{
+					fmt.Sprintf("prctl;native;%s,99999", arg),
+					"read;native;",
+				} {
+					simulateBpf(c, seccompWhitelist, bad, main.SeccompRetKill)
+				}
+			}
+		}
+	}
+}
```

Comment:
I'd like to see syntax checks for each item in the resolver map (which is what we're doing with current C implementation). As such:

* Please implement TestRestrictionsWorkingArgsClone for test_restrictions_working_args_clone. I know you did a spread test for this one, but it should be done as a unit test instead.
* Please implement TestRestrictionsWorkingArgsMknod for test_restrictions_working_args_mknod.
* Please implement TestRestrictionsWorkingArgsPrio for test_restrictions_working_args_prio.
* Please implement TestRestrictionsWorkingArgsTermios for test_restrictions_working_args_termios

I know with some of these they are tested in TestCompile, but it seems clearer to have the args all in one place rather than spread between two tests.